### ISSUE TITLE
preinstall and postinstall are run twice

### ISSUE THREAD

**Issue by andreineculau** [pnpm/pnpm#856]
### Code to reproduce the issue:

see https://github.com/andreineculau/package-json-scripts

```shell
git clone https://github.com/andreineculau/package-json-scripts
pnpm install # lists preinstall, preinstall, install, postinstall, postinstall, ...

npm install # lists preinstall, install, postinstall, ...
```

### Additional information:

Regression after https://github.com/pnpm/pnpm/issues/854

`npm run install` actually calls preinstall and postinstall on its own :(
See https://github.com/npm/npm/blob/latest/lib/run-script.js#L157

**Comment by zkochan** [comment]
oh, indeed, thanks. I'll fix it soon

**Comment by zkochan** [comment]
Fixed published in [v1.8.1](https://github.com/pnpm/pnpm/releases/tag/v1.8.1)

**Comment by andreineculau** [comment]
confirmed that it works as intended now. Thanks (even for the "beautiful" hack)!

PS: I've opened up an issue with npm on the topic https://github.com/npm/npm/issues/17901
